heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerHttpServer.java

```diff
@@ -0,0 +1,235 @@
+   * __jvm-memory-mb-total __jvm-gc-collection-time-ms __server/__time_spent_back_pressure_initiated \
+   * __time_spent_back_pressure_by_compid
+   */
+  public static void main(String[] args)
+      throws ExecutionException, InterruptedException, IOException {
+    if (args.length < 3) {
+      System.out.println(
+          "Usage: java MetricsQuery <topology_name> <component_name> <metrics_name>");
+    } else {
+      System.out.println("topology: " + args[0] + "; component: " + args[1]);
+    }
+
+    Config config = Config.newBuilder()
+        .put(Key.STATEMGR_ROOT_PATH,
+            System.getProperty("user.home") + "/.herondata/repository/state/local")
+        .build();
+    LocalFileSystemStateManager stateManager = new LocalFileSystemStateManager();
+    stateManager.initialize(config);
+
+    TopologyMaster.MetricsCacheLocation location =
+        stateManager.getMetricsCacheLocation(null, args[0]).get();
+    if (location == null || !location.isInitialized()) {
+      System.out.println("MetricsCacheMgr is not ready");
+      return;
+    }
+
+    // construct metric cache stat url
+    String url = "http://" + location.getHost() + ":" + location.getStatsPort()
+        + MetricsCacheManagerHttpServer.PATH_STATS;
+
+    // construct query payload
+    byte[] requestData = TopologyMaster.MetricRequest.newBuilder()
+        .setComponentName(args[1])
+        .setMinutely(true)
+        .setInterval(-1)
+        .addAllMetric(Arrays.asList(Arrays.copyOfRange(args, 2, args.length)))
+        .build().toByteArray();
+
+    // http communication
+    HttpURLConnection con = NetworkUtils.getHttpConnection(url);
+    NetworkUtils.sendHttpPostRequest(con, "X", requestData);
+    byte[] responseData = NetworkUtils.readHttpResponse(con);
+
+    // parse response data
+    TopologyMaster.MetricResponse response = TopologyMaster.MetricResponse.parseFrom(responseData);
+
+    //print result
+    System.out.println(response.toString());
+  }
+
+  public void start() {
+    server.start();
+  }
+
+  public void stop() {
+    server.stop(0);
+  }
+
+  // T - request, U - response
+  abstract class RequestHandler<T extends GeneratedMessage, U extends GeneratedMessage>
+      implements HttpHandler {
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+      // get the entire stuff
+      byte[] payload = NetworkUtils.readHttpRequestBody(httpExchange);
+      T req;
+      try {
+        req = parseRequest(payload);
+      } catch (InvalidProtocolBufferException e) {
+        LOG.log(Level.SEVERE,
+            "Unable to decipher data specified in Request: " + httpExchange, e);
+        httpExchange.sendResponseHeaders(400, -1); // throw exception
+        return;
+      }
+      // query cache
+      U res = generateResponse(req, metricsCache);
+      // response
+      NetworkUtils.sendHttpResponse(httpExchange, res.toByteArray());
+      // close
```

Comment:
please review all comments in the PR to verify that they are all helpful and that without them, the code would be unclear. For example, comments like "print result" above and "close" below are unnecessary:
```
// close
httpExchange.close();
```